arch/x86/core/x86_mmu.c

```diff
@@ -990,12 +990,12 @@ static void setup_thread_tables(struct k_thread *thread,
 
 			/* Pulled out of reserved memory in the stack object */
 			user_table = thread_page_pool_get(thread);
-			__ASSERT(user_table != NULL, "out of memory")
+			__ASSERT(user_table != NULL, "out of memory");
 
 			(void)memcpy(user_table, master_table,
 				     table_size(level));
 
-			*link = ((pentry_t)user_table) | INT_FLAGS;
+			*link = (*(pentry_t *)user_table) | INT_FLAGS;
```

Comment:
Tentative fix for below warning. Not sure if this is correct, I may be totally misunderstanding what the code around it is doing.
```
warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  998 |    *link = ((pentry_t)user_table) | INT_FLAGS;
```

Reply:
this is very subtle.

page tables are a type of radix tree where the leaf nodes contain the virtual-to-physical mapping, but the intermediate nodes just link to other tables -- by physical address.

here what we want to put in `*link` is the physical address of user table plus the bits in INT_FLAGS (which are the page table flags we use for all non-leaf tables).

user_table is a 32-bit virtual address. but by casting it to pentry_t, pentry_t if CONFIG_X86_PAE is enabled is a 64-bit physical address. they are otherwise the same value, but that's what the warning comes from.

this is all somewhat complicated and unintuitive but the key thing is we don't want to dereference `user_table` here, we want to preserve its pointer value.

so maybe:

```
*link = ((pentry_t)(uintptr_t)user_table) | INT_FLAGS;
```

Will make the compiler happy.